graalpy/graalpy-scripts-debug-guide/README.md

```diff
@@ -0,0 +1,244 @@
+We will demonstrate on both build systems.
+
+## 4.1 Dependency configuration
+
+We have added the required dependencies for GraalPy in the `<dependencies>` section of the POM or to the `dependencies` block in the `build.gradle.kts` file.
+
+`pom.xml`
+```xml
+<dependency>
+    <groupId>org.graalvm.polyglot</groupId>
+    <artifactId>python</artifactId> <!-- ① -->
+    <version>24.1.1</version>
+    <type>pom</type> <!-- ② -->
+</dependency>
+<dependency>
+    <groupId>org.graalvm.polyglot</groupId>
+    <artifactId>polyglot</artifactId> <!-- ③ -->
+    <version>24.1.1</version>
+</dependency>
+<dependency>
+    <groupId>org.graalvm.tools</groupId>
+    <artifactId>dap-tool</artifactId> <!-- ④ -->
+    <version>24.1.1</version>
+</dependency>
+```
+
+`build.gradle.kts`
+```kotlin
+implementation("org.graalvm.polyglot:python:24.1.1") // ①
+implementation("org.graalvm.polyglot:polyglot:24.1.1") // ③
+implementation("org.graalvm.tools:dap-tool:24.1.1") // ④
+```
+
+❶ The `python` dependency is a meta-package that transitively depends on all resources and libraries to run GraalPy.
+
+❷ Note that the `python` package is not a JAR - it is simply a `pom` that declares more dependencies.
+
+❸ The `polyglot` dependency provides the APIs to manage and use GraalPy from Java.
+
+❹ The `dap` dependency provides a remote debugger for GraalPy that we can use when Python code is embedded in a Java application.
+
+## 4.2 Adding a Python script
+
+We can just include simple Python scripts in our resources source folder.
+In this example, the script contains a function that uses the Python standard library to compute the similarity between two files.
+
+`src/main/resources/compare_files.py`
+```python
+import polyglot  # pyright: ignore
+
+from difflib import SequenceMatcher
+from os import PathLike
+
+
+@polyglot.export_value # ①
+def compare_files(a: PathLike, b: PathLike) -> float:
+    with open(a) as file_1, open(b) as file_2: 
+        file1_data = file_1.read() 
+        file2_data = file_2.read() 
+        similarity_ratio = SequenceMatcher(None, file1_data, file2_data).ratio()
+        return similarity_ratio
+```
+
+❶ The only GraalPy-specific code here is this `polyglot.export_value` annotation, which makes the function accessible by name to the Java world.
+
+## 4.2.1 Working with GraalPy in VSCode
+
+You can use [pyenv](https://github.com/pyenv/pyenv) or [pyenv-win](https://github.com/pyenv-win/pyenv-win) with the [Python extensions](https://marketplace.visualstudio.com/items?itemName=ms-python.python) in VSCode to setup and use GraalPy during development.
+You can than edit and debug your Python files using the standard Python tooling.
+
+![Gif animation of installing GraalPy with pyenv](./graalpy-vscode-pyenv.gif)
+![Gif animation of using GraalPy in VSCode](./graalpy-vscode-select.gif)
+![Gif animation of debugging with GraalPy in VSCode](./graalpy-vscode-debug.gif)
```

Comment:
I am getting this:
![image](https://github.com/user-attachments/assets/b3d34081-051e-45d1-b1db-344a83a7f600)
have you seen anything like that before?

Reply:
Argh. They haven't pushed the new version out, yet, and I still had my local changes that I pushed upstream. They updated the debugpy version in the extension to one with the fix (https://github.com/microsoft/vscode-python-debugger/pull/505) but haven't pushed a new extension release